contributors/design-proposals/scheduling/coscheduling.md

```diff
@@ -0,0 +1,174 @@
+   * allocate (but not bind) resource to Pods according to Pod's spec, e.g. `NodeAffinity`
+   * bind all Pods to hosts until job ready (`numMember` <= `allocated Pods` + `succeeded Pods`)
+4. If can not allocate enough resources to the job, the pods are keeping pending; and the resource can not allocate to other job
+
+That may make resources (less than job's resource request) idle for a while, e.g. a huge job. The solution, e.g. backfill other smaller jobs to improve the resource utilization, will be proposed in coming release. In `allocate` phase, only pod's `NodeAffinity` takes effect; the other predicates/priorities will be included on-demand in coming release.
+
+### Customized Controller
+
+A typical example of customized controller is [kubeflow/tf-operator](https://github.com/kubeflow/tf-operator), which managed the Pods for TensorFlow on Kubernetes, required `gang-scheduling` in upstream. Here's an example of customized controller that demonstrated the usage of  `gang-scheduling` in `kube-arbitrator`.
+
+Usually, CRD ([CustomResourceDefinitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)) feature is used to introduce a customized **Kind**, named `CRDJob` as example. The customized controller, named `CRDJobController`, watches it and manages lifecycle of it:
+
+1. For each `CRDJob`, `CRDJobController` creates a `CRDJob` and `PodGroup` (one `CRDJob` with one `PodGroup` as example). The attributes of `PodGroup`should be set accordingly, e.g `numMember`; it's up to customized controller on how to manage relationship between `PodGroup` and `CRDJob`, e.g. `metadata.name`.
+2. When `CRDJobController` create Pods, its annotation should be set accordingly. `kube-arbitrator` follows gang-scheduling logic to schedule those pods in batch.
+3. When pods failed/deleted/unschedulable, it up to `CRDJobController` on how to manage `CRDJob`'s lifecycle. For example, if `CRDJobController` manages lifecycle itself, set `.spec.Policy` of `PodGroup` to nil; otherwise, `PodGroupController` will manage the lifecycle as described above.
+4. If `CRDJob` was deleted, the `PodGroup` must be deleted accordingly.
+
+## Feature Interaction
+
+### Multi-scheduler
+
+When multi-scheduler was enabled, there maybe decision conflict between different schedulers; and the kubelet will reject one pod (failed) if conflict. The controller will handle rejected pods based on its lifecycle policy for failed pods. It up to cluster admin to avoid such kind of conflict, e.g. node affinity which is supported by kube-arbitrator in 1.13.
+
+### Priority/Preemption
+
+As described in `multi-scheduler` section, it's better to run batch and service workload in different zone to avoid confliction.
+Rejected a batch/run-to-complete pod may trigger the restarting of the whole `PodGroup` which impact the performance. The solution on how to handle confliction better will be proposed in coming release.
```

Comment:
s/Rejected a batch/run-to-complete pod/A preempted batch/run-to-completion Pod/

Reply:
hm.. I think it should be "rejected" which means rejected by kubelet.